moml/models/mgnn/evaluation/metrics.py

```
@@ -35,347 +60,625 @@
-    r2 = r2_score(true_values, pred_values)
-    if r2 < 0:
-        r2 = 0.0
+    except ValueError:
+        # Re-raise ValueError as-is (for empty arrays, NaN/Inf validation)
+        raise
+    except TypeError:
+        # Re-raise TypeError as-is (for unsupported types)
+        raise
+    except Exception as e:
+        logger.error(f"Failed to convert {name} to numpy array: {e}")
+        raise TypeError(f"Could not convert {name} to numpy array: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)